presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -317,7 +317,7 @@ private static PlanNode planTableScan(
                             node.getOutputSymbols(),
                             node.getAssignments(),
                             Optional.of(layout.getLayout().getHandle()),
-                            newDomain.intersect(layout.getLayout().getPredicate()),
```

Comment:
that intersection was odd for me. Shouldn't we simplify `layout.getLayout().getPredicate()` so that it doesn't explode when it's converted back to predicate in `com.EffectivePredicateExtractor.Visitor#visitTableScan`

Reply:
> Those predicate would disappear from the filter above the TableScan.

I don't see why it would happen. Let's check:
1. When filter was on top it was converted to
** exact tuple domain (for expressions that translate directly to tuple domain). Let's call it `FT`
** remaining expressions, `EX`
2. Connector was asked to enforce `FT`. It returned some tuple domain `GT` that was generalization of rows that connector returned. It also returned unenforced domain `UX`.
3. `currentConstraint` was set to intersection of `FT` and `GT` (ok, produced rows could exceed that domain), but then `UX AND EX` should always be applied to rows that exceeded `FT`. Therefore predicate shouldn't dispensary.

Reply:
Also when connector returned generalizing tuple domain but no unenforced predicates, then it seems that intersection was needed.

Reply:
To clarify, filters will only disappear if the intersection was removed without making other necessary changes.

The necessary changes were made in d54521f24ebf49754da2833f748d8fb2d4149cc7. And I intended to remove the intersection in that commit. However, the intersection was not removed due to my mistake while addressing comments.

(just to provide context to readers. @sopel39 and I resolved the confusion offline.)